BZ2028564:adding CLI procedure for deploying kmstate operator

skrthomas opened the pull request:
Version: 4.10+

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2028564

Preview: https://deploy-preview-43723--osdocs.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-about-the-k8s-nmstate-operator.html

Details: Adding a procedure for installing/deploying the nmstate operator via CLI.

achuzhoy commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,125 @@
+
+. Create the namespace for the nmstate operator:
++
+[source,yaml]
```
cat << EOF | oc apply -f -

achuzhoy commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,125 @@
+  finalizers:
+  - kubernetes
+----
+
```
EOF

opayne1 commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,104 @@
+
+.Procedure
+
+. Create the nmstate operator namespace:
```
Should this be `nmstate`?

opayne1 commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,104 @@
+EOF
+----
+
+. Create the OperatorGroup:
```
Should this be `OperatorGroup`?

opayne1 commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,104 @@
+  - openshift-nmstate
+EOF
+----
+. Subscribe to the nmstate Operator:
```
s/Operator/operator

opayne1 reviewed (commented):
I think `nmstate` and `OperatorGroup` need to be in backticks. Other than that just one little nit on our end. Lgtm pending the QA ack.

skrthomas commented:
@zhaozhanqi can you PTAL at this PR for QA? 
@achuzhoy if you could also take another look, that would be fantastic!

Thank you in advance :)

skrthomas commented:
@amalykhi thanks for your comments! I removed that node selector and adjusted the versions on the other comments you left. Since nmstate operator only GA'd in 4.10, I'll only publish this CLI command to 4.10+.

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+:_content-type: PROCEDURE
+[id="installing-the-kubernetes-nmstate-operator-CLI_{context}"]
+= Installing the Kubernetes NMState Operator using the CLI
+Instead of using the web console, you can install the Kubernetes NMState Operator by using the {product-title} CLI (oc).
```
Wondering if you have any thoughts about omitting the introductory clause: 

You can install the Kubernetes NMState Operator by using the {product-title} CLI (oc) instead of using the web console.

Or simply: 

You can install the Kubernetes NMState Operator by using the {product-title} CLI (oc).

Or

You can also install the Kubernetes NMState Operator by using the {product-title} CLI (oc).

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
```
Wouldn't these be past tense since they're prerequisites? I know we often write OC CLI has past tense, anyways:

"* You have installed the OpenShift CLI (`oc`)."

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+* Install the OpenShift CLI (`oc`).
+
+* Log in as a user with `cluster-admin` privileges.
```
Would we make the other past tense? This is a common occurrence: 

"* You must be logged in as a user with `cluster-admin` privileges."

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+.Procedure
+
+. Create the `nmstate` operator namespace:
```
Operator (https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization)

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+EOF
+----
+
+. Create instance of the `nmstate` operator:
```
Operator

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+----
+
+.Verification
+Confirm that the deployment for the `nmstate` operator is running:
```
Operator

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+. Create the `nmstate` operator namespace:
++
+[source,terminal]
```
[source,yaml]

skrthomas replied:
So for these, I was also confused how to tag the codeblocks. The SME gave me this [MetalLB Operator cli install](https://docs.openshift.com/container-platform/4.9/networking/metallb/metallb-operator-install.html#nw-metallb-installing-operator-cli_metallb-operator-install) as a reference for this nmstate Operator, and in the adoc file these yaml looking pieces are [source,terminal], not yaml. So...I did the same. There is the cat command piece too, so maybe that's why its terminal?

stevsmit replied:
You're right.

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+. Create the `OperatorGroup`:
++
+[source,terminal]
```
[source,yaml]

stevsmit commented on modules/k8s-nmstate-deploying-nmstate-CLI.adoc:
```diff
@@ -0,0 +1,101 @@
+
+. Create instance of the `nmstate` operator:
++
+[source,terminal]
```
[source,yaml]

stevsmit commented:
/lgtm

skrthomas commented:
/cherry-pick enterprise-4.10

skrthomas commented:
/cherry-pick enterprise-4.11
